Drop changes on finalize in postprocessor and user object

fdkong opened the pull request:
Refs #12948

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

fdkong committed:
Drop changes on finalize in postprocessor and user object

Refs #12948

permcody commented on test/tests/transfers/multiapp_conservative_transfer/tests:
```diff
@@ -4,6 +4,7 @@
   [./conservative_transfer]
     type = Exodiff
     input = master_conservative_transfer.i
+    max_parallel = 6
```
This isn't the right fix - We should probably put the standard "errorIfParallel" stuff in this transfer until we can look at why it doesn't work for DistributedMesh. I'll accept this for now, but it's not clear to the user why you are limiting to 6 (other than you are hiding that it's actually broken).

fdkong replied:
It is for the integration. I will fix it soon.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/340961/) on fb00656 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13451/site)

This comment will be updated on new commits.

fdkong committed:
Use MultiAppMeshFunctionTransfer instead of MultiAppNearestNodeTransfer

Two reasons:

(1) MultiAppMeshFunctionTransfer makes more sense in the conservative transfer

(2) MultiAppNearestNodeTransfer is unstable in parallel (we should fix this in a different PR)

fdkong commented:
@permcody  You may need to Review this PR again. I use `MultiAppMeshFunctionTransfer` instead of `MultiAppNearestNodeTransfer` in master_conservative_transfer.i

fdkong commented:
`MultiAppNearestNodeTransfer` is unstable in parallel.
